inference/core/interfaces/camera/camera.py

```diff
@@ -60,36 +60,47 @@ def start(self):
 
     def update(self):
         """Update the frame by reading from the webcam."""
+        frame_id = 0
+        skip_seconds = 0
+        last_frame_position = time.perf_counter()
+        t0 = time.perf_counter()
         while True:
             t1 = time.perf_counter()
             if self.stopped is True:
                 break
-            self.grabbed, self.frame = self.vcap.read()
-            if self.frame is not None and self.grabbed:
-                self.frame_id += 1
-                self.pil_image = Image.fromarray(
-                    cv2.cvtColor(self.frame, cv2.COLOR_BGR2RGB)
-                )
+
+            self.grabbed = self.vcap.grab()
+            if self.grabbed:
+                frame_id += 1
+                if (
+                    self.enforce_fps != "skip"
+                    or t1 >= last_frame_position + skip_seconds
+                ):
+                    ret, self.frame = self.vcap.retrieve()
+                    logger.debug("video capture FPS: %s", frame_id / (t1 - t0))
+                    if self.frame is not None:
+                        last_frame_position = t1
+                        self.frame_id = frame_id
+                    else:
+                        logger.debug("[Exiting] Frame not available to retrieve")
+                        self.stopped = True
+                        break
```

Comment:
I think you may want something like:
```suggestion
                    ret, frame = self.vcap.retrieve()
                    logger.debug("video capture FPS: %s", frame_id / (t1 - t0))
                    if frame is not None:
                        last_frame_position = t1
                        self.frame_id = frame_id
                        self.frame = frame
                    else:
                        logger.debug("[Exiting] Frame not available to retrieve")
                        self.stopped = True
                        break
```

As is, there is a risk that self.frame and self.frame_id become out of sync, or that self.frame_id is defined, but self.frame is None.

Reply:
Great catch. Committing!